### PR TITLE
Language Selection

### PR DESCRIPTION
**Pull request by ChrisHuynh333**
## What does this PR do and why?
In this PR, locale selection has been implemented under the user's profile preferences settings. This will allow users to select between English and French language translations for the website.

## Screenshots or screen recordings
image
image

## How to set up and validate locally
1. Navigate to profile preferences.
2. Under language, select either English or French and submit.
- Only the language card contains translations currently. Therefore, the language card's wording should be translated based on the language selected.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

**Comment by joshsadam** [comment]
I messaged you in teams, but want to post it here as well. Can you update the UI to:

image

You can see how to use sections in the card component in LookBooK: [http://localhost:3000/rails/lookbook/inspect/viral\_card/with\_multiple\_sections](http://localhost:3000/rails/lookbook/inspect/viral_card/with_multiple_sections)

Can you also make the language update as soon as the radio button is clicked please, like in the colour theme

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
This is looking good and working as expected. Just a few small things below

File: app/controllers/profiles/preferences_controller.rb [hunk 1]
```diff
@@ -11,5 +11,28 @@ def show
...
+
+      respond_to do |format|
+        if @user.update(update_params)
+          flash[:success] = t('.success')
+          format.html { redirect_to profile_preferences_path }
+        else
+          format.html { render :show, status: :unprocessable_entity, locals: { user: @user } }
+        end
+      end
+      locale = current_user.try(:locale) || I18n.default_locale
```
[the remark this answers is not in the document]
Comment (ChrisHuynh333) [reply app/controllers/profiles/preferences_controller.rb]: Fixed in 0033252d0f2799b90d626c5c1b9af2d462b9234e

File: config/application.rb [hunk 2]
```diff
@@ -51,5 +51,11 @@ class Application < Rails::Application
         end
       }
     end
+
+    # Only enables locale from rails_i18n. Other options include ordinals, pluralization, transliteration
+    # https://github.com/svenfuchs/rails-i18n
+    config.rails_i18n.enabled_modules = [:locale]
+    # Only enables en and fr locales, avoiding unnecessarily loading other locales
+    config.i18n.available_locales = %i[en fr]
```
[the remark this answers is not in the document]
Comment (ChrisHuynh333) [reply config/application.rb:59]: Fixed in 0033252d0f2799b90d626c5c1b9af2d462b9234e

File: config/initializers/locale.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+# config/initializers/locale.rb
+
+# Where the I18n library should search for translation files
+I18n.load_path += Dir[Rails.root.join('config/locales/*.{rb,yml}')]
+
+# Permitted locales available for the application
+I18n.available_locales = %i[en fr]
```
[the remark this answers is not in the document]
Comment (ChrisHuynh333) [reply config/initializers/locale.rb]: Fixed in 0033252d0f2799b90d626c5c1b9af2d462b9234e

**Comment by ChrisHuynh333** [comment]
> I messaged you in teams, but want to post it here as well. Can you update the UI to:
> 
> image
> 
> You can see how to use sections in the card component in LookBooK: http://localhost:3000/rails/lookbook/inspect/viral_card/with_multiple_sections
> 
> Can you also make the language update as soon as the radio button is clicked please, like in the colour theme

I've fixed the cards in 0033252d0f2799b90d626c5c1b9af2d462b9234e
And the language selection now submits the form onchange in fb9058b1930ca2f73b5332f50a577648ee5c73a7

File: app/views/profiles/preferences/_locale_form.html.erb [hunk 4]
```diff
@@ -0,0 +1,27 @@
+
```
Comment (deepsidhu85) [inline app/views/profiles/preferences/_locale_form.html.erb:1]: Can you remove these 2 blank lines at the beginning of the file

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
Looks good to me! Just a comment below about removing blank lines

**Comment by joshsadam** [comment]
Can you add a French translation for the update success, this just kind of looked funny.

File: test/system/profile_test.rb [hunk 5]
```diff
@@ -103,4 +103,17 @@ def setup
...
+    visit profile_preferences_path
+
+    assert_text I18n.t(:'profiles.preferences.locale_form.en')
+  end
+  test 'can update language to french' do
+    visit profile_preferences_path
+
+    find('input[id=user_locale_fr]').click
+
+    assert_text 'Anglais'
```
Comment (deepsidhu85) [inline test/system/profile_test.rb:117]: I missed one thing sorry, can you check that the flash alert displays when the locale is updated, and check for the success text within it

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
Sorry missed a comment for the testing

File: test/system/profile_test.rb [hunk 6]
```diff
@@ -103,4 +103,17 @@ def setup
     assert_text I18n.t(:'profiles.personal_access_tokens.index.active_personal_access_tokens',
                        count: @active_token_count - 1)
   end
+
+  test 'can view language selection' do
+    visit profile_preferences_path
+
+    assert_text I18n.t(:'profiles.preferences.locale_form.en')
+  end
```
Comment (joshsadam) [inline test/system/profile_test.rb:111]: **nitpick:** Can you add aline break between these two tests please.

**Comment by ChrisHuynh333** [comment]
> Can you add a French translation for the update success, this just kind of looked funny.
> 
> 

Fixed in 40a49b116b2e7f0f55b16806e4208811e3333b02

File: test/system/profile_test.rb (hunk 5 above)
Comment (ChrisHuynh333) [reply test/system/profile_test.rb]: Fixed in 40a49b116b2e7f0f55b16806e4208811e3333b02

File: test/system/profile_test.rb (hunk 6 above)
Comment (ChrisHuynh333) [reply test/system/profile_test.rb:111]: Fixed in 9f4ef54c2cff211eacd6f2db2111499a09aff0d9

File: app/views/profiles/preferences/_locale_form.html.erb (hunk 4 above)
Comment (ChrisHuynh333) [reply app/views/profiles/preferences/_locale_form.html.erb]: Fixed in 9f4ef54c2cff211eacd6f2db2111499a09aff0d9

**Review by deepsidhu85 — APPROVED** [review]

**Pull request merged**